services/QuillLMS/app/workers/admin_diagnostic_reports/enqueue_csv_email_worker.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module AdminDiagnosticReports
+  class EnqueueCsvEmailWorker
+    include Sidekiq::Worker
+
+    BASE_REPORT_NAME = 'diagnostic_growth_report'
+    OVERVIEW_REPORT_NAME = 'diagnostic_growth_report_overview'
+    SKILL_REPORT_NAME = 'diagnostic_growth_report_skill'
+    STUDENT_REPORT_NAME = 'diagnostic_growth_report_student'
+
+    DEFAULT_TIMEFRAME = 'this-school-year'
+    DEFAULT_AGGREGATION = 'grade'
+    DEFAULT_DIAGNOSTIC_ID = 1663
+
+    def perform(user_id)
```

Comment:
I haven't finished reviewing, but my initial question is: why break the async work into two workers? I get that both working are doing a lot of stuff, but that complexity can be managed with plain Ruby classes, rather than multiple workers. 

Related: If we're keeping two workers, the name should be more expansive. As in,  `PreprocessFiltersAndEnqueue*` vs. `Enqueue*`

Reply:
This is a fair question.  I don't think it has to be split into two parts, but I'm partially anticipating a future feature (which may be a mistake at this point).  Specifically, the plan is to eventually add the ability for admins to subscribe to regular emails of these reports.  I expect that if we do that, then we'll actually be using a different set of saved filters, so separating the logic of extracting and formatting filters from the actual CSV generation made sense to me.  Open to the idea that we should just wait until building that future feature before worrying about the split.

Reply:
I think it's best to not split it for now. Reasoning:

- Outside of the context of this PR diff, the intention of the code is harder to follow when it's split across two jobs
- every job adds additional overhead of managing idempotency and failure/retry states